src/model/page-config-resolver.js

```diff
@@ -268,16 +283,22 @@ class JsonLdParser {
 
   /**
    * @param {!Object} json
-   * @param {string} expectedType
+   * @param {Array<string>} expectedTypes
    * @return {boolean}
    */
-  checkType_(json, expectedType) {
+  checkType_(json, expectedTypes) {
```

Comment:
could we export checkType() out of this and MicrodataParser method? Seems like duplicate functionality except for the argument.
In this class, isValidType() which takes two arrays. Here it gets invoked with this.valueArray_(json, '@type') as argument and in MicrodataParser, it gets itemtype.split(/\s+/)

It seems like types allowed and its management happens outside both JsonLdParser and MicrodataParser

Seems like this guidance on what's okay and not okay is coming outside of these Parsers, which decides what is considered valid configuration. Might be an overkill at this point to have a class that dictates our standard clearly (that would include type checking as well as allowed types). But exporting checkType() out of these classes will atleast consolidate common behavior into one place and be clear that these parsers take guidance from outside. Thoughts?

Reply:
done